source/misc/bash_startup.in

```diff
@@ -1,257 +1,95 @@
-#!/bin/bash
-# This is based on "preexec.bash" but is customized for iTerm2.
+# This is inspired by "preexec.bash" but is customized for iTerm2. Sourced, so
+# no shebang. This module requires 2 bash features you must not otherwise be
+# using: the DEBUG trap and the PROMPT_COMMAND variable.
 
-# Note: this module requires 2 bash features which you must not otherwise be
-# using: the "DEBUG" trap, and the "PROMPT_COMMAND" variable.  iterm2_preexec_install
-# will override these and if you override one or the other this _will_ break.
+osc='\033]'; cso='\007' # Beginning, ending escape sequences.
 
-# This is known to support bash3, as well as *mostly* support bash2.05b.  It
-# has been tested with the default shells on MacOS X 10.4 "Tiger", Ubuntu 5.10
-# "Breezy Badger", Ubuntu 6.06 "Dapper Drake", and Ubuntu 6.10 "Edgy Eft".
+# tmux / screen are not supported; I think this is the theoretical tmux hack?
+[ $TERM = screen ] && osc='\033Ptmux;\033\033]' && cso='\007\033\\'
```

Comment:
@gnachman 
> This would seem to handle an empty $TERM better, but I'm not sure what benefit the * confers.
Well, I have seen variants like `screen-256color`, so I get that part. But does having an empty `$TERM` actually happen? I suppose you *can* unset it, but I'd think bad things would start to happen pretty quickly.

@docwhat 
You're right in that 1) its not portable (and that it would need `[[ ]]`), and testing $TMUX is a good idea, probably more reliable. But there's a lot of extraneous code there.

Also, while its extremely common, `if [ condition ]; then` is redundant, and `if then`'s are one of the slowest structures in shell scripts. There are rare occasions when you need it but generally, (like in case), all you need is:

```sh

[ -z "$TMUX" ] && case "$TERM" in
  screen*) is_tmux_or_screen=t ;; # or, in our use case here, do nothing
  *|'') is_tmux_or_screen=f ;; # or, in our case, put our commands here
esac